src/ui/public/vislib/lib/axis.js

```diff
@@ -0,0 +1,228 @@
+import d3 from 'd3';
+import _ from 'lodash';
+import $ from 'jquery';
+import ErrorHandlerProvider from 'ui/vislib/lib/_error_handler';
+import AxisTitleProvider from 'ui/vislib/lib/axis_title';
+import AxisLabelsProvider from 'ui/vislib/lib/axis_labels';
+import AxisScaleProvider from 'ui/vislib/lib/axis_scale';
+
+export default function AxisFactory(Private) {
+  const ErrorHandler = Private(ErrorHandlerProvider);
+  const AxisTitle = Private(AxisTitleProvider);
+  const AxisLabels = Private(AxisLabelsProvider);
+  const AxisScale = Private(AxisScaleProvider);
+  const defaults = {
```

Comment:
in my opinion it would make it more confusing ... maybe not with the axis itself ... but AxisLabels have their own config ... following same pattern we would then need AxisLabelsOptions as well ... and AxisTitleOptions ...  and in the future i hope to do similar update to other parts of vislib where this would make even less sense.  i think by defining defaults like this its quite clear what all the options that this class can handle are. let me know if you dissagree